compiler/rustc_next_trait_solver/src/resolve.rs

```diff
@@ -22,8 +22,20 @@ where
     cache: DelayedMap<I::Ty, I::Ty>,
 }
 
+pub fn eager_resolve_vars<D: SolverDelegate, T: TypeFoldable<D::Interner>>(
```

Comment:
now that we actually check for `has_infer` in `fold_clauses`, this change may be unnecessary again :thinking:

Reply:
🤷 idc